lib/krikri/harvesters/ia_harvester.rb

```diff
@@ -0,0 +1,160 @@
+require 'uri'
+require 'json'
+require 'async_uri_getter'
+
+module Krikri::Harvesters
+  ##
+  # A harvester implementation for Internet Archive
+  #
+  class IaHarvester
+    include Krikri::Harvester
+
+    DEFAULT_THREAD_COUNT = 10
+    DEFAULT_MAX_RECORDS = 0
+
+    DOWNLOAD_BASE_URI = 'http://archive.org/download'
+
+    ##
+    # Initialize, and set default options as appropriate.
+    #
+    # @param opts [Hash] a hash of options as defined by {.expected_opts}
+    #
+    # @example
+    #    Typical instantiation, good for most cases:
+    #
+    #      Krikri::Harvesters::IaHarvester.new(uri: 'http://archive.org/...')
```

Comment:
A search URI for a collection. I found this in:
https://github.com/dpla/ingestion/blob/develop/profiles/ia.pjs

An example I used while testing:
http://archive.org/advancedsearch.php?q=collection%3A%22bostonpubliclibrary%22&fl%5B%5D=identifier&output=json